File: test-api-connection.py
import requests
import json

def test_api_connection():
    try:
        print("测试API连接...")
        
        # 测试直接连接到API服务器 (8787端口)
        print("1. 测试直接连接到API服务器 (8787端口)")
        direct_response = requests.get('http://localhost:8787/api/test-jwt-secret')
        print(f"直接连接状态码: {direct_response.status_code}")
        print(f"直接连接响应头: {direct_response.headers.get('content-type')}")
        
        if 'application/json' in direct_response.headers.get('content-type', ''):
            direct_data = direct_response.json()
            print(f"直接连接结果: {json.dumps(direct_data, indent=2, ensure_ascii=False)}")
        else:
            print(f"直接连接返回了非JSON内容，长度: {len(direct_response.text)} 字符")
        
        # 测试通过前端代理连接 (3000端口)
        print("\n2. 测试通过前端代理连接 (3000端口)")
        proxy_response = requests.get('http://localhost:3000/api/test-jwt-secret')
        print(f"代理连接状态码: {proxy_response.status_code}")
        print(f"代理连接响应头: {proxy_response.headers.get('content-type')}")
        
        if 'application/json' in proxy_response.headers.get('content-type', ''):
            proxy_data = proxy_response.json()
            print(f"代理连接结果: {json.dumps(proxy_data, indent=2, ensure_ascii=False)}")
        else:
            print(f"代理连接返回了非JSON内容，长度: {len(proxy_response.text)} 字符")
            
    except Exception as e:
        print(f"测试出错: {e}")

if __name__ == "__main__":
    test_api_connection()

File: test-api-connection.js
// 测试API连接
async function testApi() {
  try {
    console.log('测试API连接...');
    
    // 直接测试API服务器
    console.log('1. 测试直接连接到API服务器 (8787端口)');
    const directResponse = await fetch('http://localhost:8787/api/test-jwt-secret');
    const directData = await directResponse.json();
    console.log('直接连接结果:', directData);
    
    // 测试通过前端代理连接
    console.log('\n2. 测试通过前端代理连接 (3000端口)');
    const proxyResponse = await fetch('http://localhost:3000/api/test-jwt-secret');
    const proxyText = await proxyResponse.text();
    console.log('代理连接响应类型:', proxyResponse.headers.get('content-type'));
    console.log('代理连接响应长度:', proxyText.length);
    
    // 如果响应是JSON格式，尝试解析
    if (proxyResponse.headers.get('content-type')?.includes('application/json')) {
      const proxyData = JSON.parse(proxyText);
      console.log('代理连接结果:', proxyData);
    } else {
      console.log('代理连接返回了非JSON内容，可能是HTML页面');
    }
  } catch (error) {
    console.log('测试出错:', error.message);
  }
}

// 运行测试
testApi();